Source/SwiftLintFramework/Rules/ProhibitedSuperRule.swift

```diff
@@ -55,15 +55,13 @@ public struct ProhibitedSuperRule: ConfigurationProviderRule, ASTRule, OptInRule
                              kind: SwiftDeclarationKind,
                              dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
         guard let offset = dictionary["key.bodyoffset"] as? Int64,
-            let name = dictionary["key.name"] as? String
-            else { return [] }
-
-        let substructure = (dictionary["key.substructure"] as? [SourceKitRepresentable]) ?? []
-        guard kind == .functionMethodInstance &&
+            let name = dictionary["key.name"] as? String,
+            kind == .functionMethodInstance &&
             configuration.resolvedMethodNames.contains(name) &&
             dictionary.enclosedSwiftAttributes.contains("source.decl.attribute.override")
             else { return [] }
 
+        let substructure = (dictionary["key.substructure"] as? [SourceKitRepresentable]) ?? []
```

Comment:
this could also be folded into the `guard` statement above since it falls back on `[]` if the conditional cast fails, which causes to return `[]` anyway